File: redesign/storefront-a11y/build-comparison.py
#!/usr/bin/env python3
"""Build comparison.html for storefront-a11y (Stage 6).

Three before/after pairs (desktop 1920x1080, tablet 768x1024, mobile 375x812 —
both sides fullPage dsf=2), coverage table (G3c), sibling table (G3d).
Images embedded as data: URIs; the full mockup HTML is embedded in the page.
"""
import base64
import pathlib
import re

DIR = pathlib.Path(__file__).parent


def data_uri(rel: str) -> str:
    raw = (DIR / rel).read_bytes()
    return "data:image/png;base64," + base64.b64encode(raw).decode()


def table_of_64(bit: int) -> int:
    return 0


PAIRS = [
    ("Desktop — 1920×1080", "current/desktop.png", "proposal/storefront-render-desktop.png"),
    ("Tablet — 768×1024", "current/tablet.png", "proposal/storefront-render-tablet.png"),
    ("Mobile — 375×812", "current/mobile.png", "proposal/storefront-render-mobile.png"),
]

pair_html = []
for title, before, after in PAIRS:
    pair_html.append(f"""
    <section class="pair">
      <h2>{title}</h2>
      <div class="row">
        <figure><figcaption>Antes (main)</figcaption><img src="{data_uri(before)}" loading="lazy"></figure>
        <figure><figcaption>Después (propuesta)</figcaption><img src="{data_uri(after)}" loading="lazy"></figure>
      </div>
    </section>""")

pairs = "\n".join(pair_html)

coverage = """
<table>
  <thead><tr><th>Finding</th><th>Severidad</th><th>Resumen</th><th>Disposición</th></tr></thead>
  <tbody>
    <tr><td>F-1</td><td>(P1)</td><td>Badge "Cerrado" text-red-600/red-50 4.36:1</td><td>addressed — texto → red-700 #b91c1c (5.87:1) en BusinessInfoSidebar.tsx:103; ver parche en mockup</td></tr>
    <tr><td>F-2</td><td>(P1)</td><td>Badge "Abierto" text-green-600/green-50 ≈3.14:1, render condicional por día</td><td>addressed — texto → green-800 #166534 (6.27:1) en BusinessInfoSidebar.tsx:95; mismo tratamiento que F-1</td></tr>
    <tr><td>F-3</td><td>(P2)</td><td>Banner construcción con gradiente amber fuera de tokens</td><td>addressed — gradiente → tinte muted plano (bg-muted/40 ≈ #f4f4f4) en app/negocio/[slug]/page.tsx:209</td></tr>
    <tr><td>F-4</td><td>(P2)</td><td>TEST_DATA.businessSlug stale — test audita 404</td><td>addressed — slug → 'tienda-electronica-la-paz' en tests/helpers/index.ts:216; sin cambio visual (fix de test)</td></tr>
    <tr><td>F-5</td><td>(P2)</td><td>Descripción banner muted sobre gradiente ≈4.27:1 borderline</td><td>addressed — texto → foreground/80 (#474747 ≈8.5:1) en page.tsx:220</td></tr>
    <tr><td>—</td><td>—</td><td>Specs tests/m4.5-*.spec.ts con slug local stale</td><td>deferred — fuera de la suite a11y; mismo root cause, decisión en Checkpoint</td></tr>
  </tbody>
</table>"""

sibling = """
<table>
  <thead><tr><th>Instancia</th><th>Disposición</th></tr></thead>
  <tbody>
    <tr><td>components/business/BusinessInfoSidebar.tsx:95 (badge "Abierto")</td><td>addressed — texto green-800 en el mismo parche que F-2</td></tr>
    <tr><td>components/business/BusinessInfoSidebar.tsx:103 (badge "Cerrado")</td><td>addressed — texto red-700, parche F-1</td></tr>
    <tr><td>components/ui/VerificationBadge.tsx:84</td><td>not-affected — green-700/-50 pasa 4.5:1</td></tr>
    <tr><td>components/demand/DemandStatusBadge.tsx:20</td><td>not-affected — amber-800/amber-100 contraste OK</td></tr>
    <tr><td>components/network/NetworkStatusBanner.tsx:14</td><td>not-affected — amber-950 muy oscuro, OK</td></tr>
    <tr><td>app/busco/page.tsx:411</td><td>not-affected — contraste OK; drift de tokens → item tinted-card-contrast-sweep</td></tr>
    <tr><td>components/home/ResilientHomePreview.tsx:116</td><td>not-affected — contraste OK; drift → sweep</td></tr>
    <tr><td>components/products/ImageUpload.tsx:445</td><td>not-affected — contraste OK; drift → sweep</td></tr>
    <tr><td>components/shared/SingleImageUpload.tsx:216</td><td>not-affected — contraste OK; drift → sweep</td></tr>
    <tr><td>components/ui/snackbar.tsx:48,55</td><td>not-affected — variantes warning con contraste OK</td></tr>
    <tr><td>tests/m4.5-mobile.spec.ts:6 + tests/m4.5-accessibility.spec.ts:7 + tests/m4.5-account-types-e2e.spec.ts:6</td><td>deferred — fuera de suite a11y; decisión en Checkpoint</td></tr>
  </tbody>
</table>"""

mockup_html = (DIR / "proposal" / "mockup.html").read_text()
mockup_b64 = base64.b64encode(mockup_html.encode()).decode()

page = f"""<!doctype html>
<html lang="es"><head><meta charset="utf-8">
<title>comparison — storefront-a11y</title>
<style>
  body {{ font-family: system-ui, sans-serif; margin: 2rem; color: #171717; }}
  h1 {{ font-size: 1.4rem; }} h2 {{ font-size: 1.1rem; margin: 1.5rem 0 .5rem; }}
  .row {{ display: grid; grid-template-columns: 1fr 1fr; gap: 1rem; }}
  figure {{ margin: 0; }} figcaption {{ font-size: .85rem; font-weight: 600; margin-bottom: .3rem; }}
  img {{ width: 100%; border: 1px solid #ddd; border-radius: 6px; }}
  table {{ border-collapse: collapse; width: 100%; font-size: .85rem; margin: .5rem 0 1.5rem; }}
  th, td {{ border: 1px solid #ddd; padding: .35rem .5rem; text-align: left; vertical-align: top; }}
  th {{ background: #f5f5f5; }}
  .note {{ background: #fef9c3; border: 1px solid #e4e4e7; padding: .6rem .8rem; border-radius: 6px; font-size: .85rem; }}
  details {{ margin: 1rem 0; }} summary {{ cursor: pointer; font-weight: 600; }}
  textarea {{ width: 100%; height: 240px; font-family: monospace; font-size: .75rem; }}
</style></head><body>
<h1>storefront-a11y — antes / después</h1>
<p>Ruta <code>/negocio/[slug]</code> — negocio auditado: <code>tienda-electronica-la-paz</code> (estado real 2026-09-15, badge "Cerrado" renderizado — lunes closed en seed).</p>
<div class="note"><strong>Defecto de render declarado:</strong> imágenes de producto en blanco en la columna "después" (lazy-load sin JS en el mock estático — mismo defecto aceptado en buscar-a11y). Defecto conocido del pipeline: badge "Abierto" no renderiza hoy (lunes closed) — parche F-2 aplicado por código, no visible en el mock.</div>
{pairs}
<h2>Cobertura de findings (G3c)</h2>
{coverage}
<h2>Patrones cross-page (G3d)</h2>
{sibling}
<h2>Mockup embebido</h2>
<p>Mock fiel sobre DOM real hidratado (scripts removidos, CSS built inlineeado, parches mínimos inline — ruta fallback <code>stitch:fallback</code>, precedente buscar-a11y loop 1).</p>
<details><summary>Ver mockup.html (embebido, auto-contenido)</summary>
<iframe id="mockframe" style="width:100%;height:70vh;border:1px solid #ddd;border-radius:6px"></iframe>
<br><a id="dl" download="mockup.html" href="data:text/html;base64,{mockup_b64}">Descargar mockup.html</a>
<script>document.getElementById('mockframe').srcdoc = atob("{mockup_b64}");</script>
</details>
</body></html>"""

(DIR / "comparison.html").write_text(page)
print("comparison.html:", f"{(DIR / 'comparison.html').stat().st_size / 1024:.0f} KB")

File: redesign/tinted-card-contrast-sweep/build-comparison.mjs
// Stage 6 comparison.html generator — 7 before/after pairs + coverage tables
// + full proposal embedded. Images as data URIs; mockups via <iframe srcdoc>.
import { readFileSync, writeFileSync } from 'fs'

const R = new URL('./', import.meta.url).pathname
const img = (p) => `data:image/png;base64,${readFileSync(R + p).toString('base64')}`
const esc = (s) => s.replace(/&/g, '&amp;').replace(/"/g, '&quot;').replace(/</g, '&lt;')

const PAIRS = [
  ['P1 · Desktop 1920×1080 — /productos/… (owner) · F-1', 'current/desktop.png', 'proposal/render-mockup-owner-notice-desktop.png', 'OwnerListingNotice: 3 párrafos muted → text-foreground/80 (4.27:1 → ≈17.8:1)'],
  ['P2 · Tablet 768×1024 — /productos/… (owner) · F-1', 'current/tablet.png', 'proposal/render-mockup-owner-notice-tablet.png', ''],
  ['P3 · Mobile 375×812 — /productos/… (owner) · F-1', 'current/mobile.png', 'proposal/render-mockup-owner-notice-mobile.png', ''],
  ['P4 · Desktop — / header, dropdown abierto · F-3', 'current/usermenu-desktop.png', 'proposal/render-mockup-usermenu-desktop.png', 'Labels "Publicaciones"/"Cuenta" 11px muted → foreground/80 (4.27:1/4.58:1 → ≈17.8:1)'],
  ['P5 · Mobile — drawer abierto (logged-out) · F-2', 'current/drawer-mobile.png', 'proposal/render-mockup-drawer-mobile.png', 'Banner CTA muted → foreground/80 (4.27:1 → ≈17.8:1)'],
  ['P6 · Desktop — /busco/publicar draft banner · F-4', 'current/draft-demand-desktop.png', 'proposal/render-mockup-draft-desktop.png', 'Subtítulo amber muted → foreground/80 (4.39:1 → ≈18.3:1). Banner muted/30 (estado) = demo inyectado debajo — misma clase swap (4.62:1 → ≈19.3:1)'],
  ['P7 · Desktop — /busco banner offline · F-5', 'current/busco-desktop.png', 'proposal/render-mockup-offline-desktop.png', 'Amber hardcodeado → border-border/60 bg-muted/40 + foreground/80 (tratamiento storefront-a11y F-3). Banner = demo inyectado (estado condicional)'],
]

const G3C = [
  ['F-1', 'addressed', 'P1-P3 (mockup-owner-notice): 3 párrafos OwnerListingNotice.tsx:23,29,34 muted→text-foreground/80'],
  ['F-2', 'addressed', 'P5 (mockup-drawer): MobileNavigationDrawer.tsx:122 muted→text-foreground/80'],
  ['F-3', 'addressed', 'P4 (mockup-usermenu): UserMenu.tsx:102,128 labels 11px muted→text-foreground/80'],
  ['F-4', 'addressed', 'P6 (mockup-draft): 6 banners en 3 archivos — subtítulos amber + estados muted/30 → text-foreground/80'],
  ['F-5', 'addressed', 'P7 (mockup-offline): ResilientHomePreview.tsx:116 + app/busco/page.tsx:411 amber→bg-muted/40 neutro'],
]

const G3D = [
  ['components/products/OwnerListingNotice.tsx', 'addressed'],
  ['components/layout/MobileNavigationDrawer.tsx', 'addressed'],
  ['components/layout/UserMenu.tsx', 'addressed'],
  ['components/products/ProductFormWizard.tsx', 'addressed'],
  ['components/demand/DemandPostForm.tsx', 'addressed'],
  ['components/profile/BusinessProfileForm.tsx', 'addressed'],
  ['components/home/ResilientHomePreview.tsx', 'addressed'],
  ['app/busco/page.tsx', 'addressed'],
  ['app/buscar/page.tsx', 'not-affected'],
  ['app/(auth)/register/page.tsx', 'not-affected'],
  ['components/products/CategoryGrid.tsx', 'not-affected'],
  ['components/products/ImageUpload.tsx', 'not-affected'],
  ['components/demand/OfferProductModal.tsx', 'not-affected'],
  ['lib/constants/productPresentation.ts', 'deferred'],
  ['components/network/NetworkStatusBanner.tsx', 'deferred'],
  ['components/demand/DemandStatusBadge.tsx', 'not-affected'],
  ['components/ui/snackbar.tsx', 'not-affected'],
  ['components/products/ProductDetailPageClient.tsx', 'not-affected'],
  ['app/page.tsx', 'not-affected'],
  ['app/perfil/demandas/page.tsx', 'not-affected'],
  ['components/profile/BusinessHoursEditor.tsx', 'not-affected'],
]
const WHY = {
  'app/buscar/page.tsx': 'CTA card ya fixed por buscar-a11y (text-foreground/80) — es la referencia visual del tratamiento',
  'app/(auth)/register/page.tsx': 'botón/campos text-foreground sobre bg-primary/5 — sin texto muted',
  'components/products/CategoryGrid.tsx': 'estado de selección (border + text-foreground)',
  'components/products/ImageUpload.tsx': 'highlight de selección, sin texto muted',
  'components/demand/OfferProductModal.tsx': 'ring de selección, sin texto muted',
  'lib/constants/productPresentation.ts': 'metaLabel 4.62:1 pasa; constante compartida con ProductCard (blanco, 4.73:1) — endurecer oscurecería superficies fuera de scope. Seguimiento en PLAN ops notes',
  'components/network/NetworkStatusBanner.tsx': 'amber text-amber-950 ≈13:1 pasa; componente global distinto — candidato limpieza propia',
  'components/demand/DemandStatusBadge.tsx': 'badge amber-800/amber-100 ≈6:1 con dark variants',
  'components/ui/snackbar.tsx': 'variantes amber con dark: pareadas ≈10:1',
  'components/products/ProductDetailPageClient.tsx': 'specs card bg-muted/30 con texto foreground (metaLabel = caso deferred)',
  'app/page.tsx': 'secciones bg-muted/30 con contenido foreground',
  'app/perfil/demandas/page.tsx': 'tab strip bg-muted/30 con texto foreground',
  'components/profile/BusinessHoursEditor.tsx': 'chip día cerrado bg-muted/30 sin texto muted',
}

const MOCKUPS = [
  ['mockup-owner-notice', 'F-1 · OwnerListingNotice (página completa del dueño)'],
  ['mockup-usermenu', 'F-3 · UserMenu dropdown abierto'],
  ['mockup-drawer', 'F-2 · Drawer móvil (banner logged-out)'],
  ['mockup-draft', 'F-4 · Banners borrador (demand wizard)'],
  ['mockup-offline', 'F-5 · Banner offline neutralizado (busco)'],
]

let html = `<!DOCTYPE html>
<html lang="es">
<head>
<meta charset="utf-8">
<title>comparison — tinted-card-contrast-sweep</title>
<style>
  body { font-family: system-ui, sans-serif; margin: 0; background: #fafafa; color: #18181b; }
  header { position: sticky; top: 0; background: #fff; border-bottom: 1px solid #e4e4e7; padding: 12px 20px; z-index: 10; }
  h1 { font-size: 18px; margin: 0 0 4px; }
  header p { margin: 0; font-size: 12px; color: #52525b; }
  main { padding: 20px; max-width: 1400px; margin: 0 auto; }
  section.pair { background: #fff; border: 1px solid #e4e4e7; border-radius: 10px; padding: 16px; margin-bottom: 24px; }
  section.pair h2 { font-size: 14px; margin: 0 0 4px; }
  section.pair .note { font-size: 12px; color: #52525b; margin: 0 0 12px; }
  .sides { display: grid; grid-template-columns: 1fr 1fr; gap: 12px; }
  .side h3 { font-size: 12px; margin: 0 0 6px; color: #52525b; text-transform: uppercase; letter-spacing: .04em; }
  .side img { width: 100%; border: 1px solid #e4e4e7; border-radius: 6px; display: block; }
  table { border-collapse: collapse; width: 100%; font-size: 12.5px; background: #fff; }
  th, td { border: 1px solid #e4e4e7; padding: 6px 10px; text-align: left; vertical-align: top; }
  th { background: #f4f4f5; }
  code { background: #f4f4f5; padding: 1px 4px; border-radius: 4px; font-size: 11.5px; }
  details { margin-bottom: 12px; }
  summary { cursor: pointer; font-weight: 600; font-size: 13px; }
  iframe { width: 100%; height: 520px; border: 1px solid #e4e4e7; border-radius: 6px; background: #fff; }
  .transparency { background: #fefce8; border: 1px solid #eab30840; border-radius: 8px; padding: 10px 14px; font-size: 12.5px; }
</style>
</head>
<body>
<header>
  <h1>tinted-card-contrast-sweep — antes / después</h1>
  <p>7 pares · F-1..F-5 · fix uniforme text-foreground/80 (17.8–19.3:1 verificado) + banner neutro bg-muted/40 · 2026-09-17</p>
</header>
<main>
`

for (const [title, before, after, note] of PAIRS) {
  html += `<section class="pair">
  <h2>${title}</h2>
  ${note ? `<p class="note">${note}</p>` : ''}
  <div class="sides">
    <div class="side"><h3>Antes (current/)</h3><img src="${img(before)}" alt="${before}"></div>
    <div class="side"><h3>Después (proposal/)</h3><img src="${img(after)}" alt="${after}"></div>
  </div>
</section>\n`
}

html += `<section>
  <h2 style="font-size:15px">G3c — cobertura de findings</h2>
  <table><tr><th> Finding </th><th> Disposición </th><th> Dónde </th></tr>
`
for (const [f, d, w] of G3C) html += `  <tr><td> ${f} </td><td> ${d} </td><td> ${w} </td></tr>\n`
html += `  </table>
</section>

<section>
  <h2 style="font-size:15px">G3d — consistencia del patrón cross-page</h2>
  <table><tr><th> Instancia </th><th> Disposición </th><th> Razón </th></tr>
`
for (const [p, d] of G3D) html += `  <tr><td><code>${p}</code></td><td> ${d} </td><td> ${WHY[p] ?? ''} </td></tr>\n`
html += `  </table>
</section>

<section>
  <h2 style="font-size:15px">Propuesta completa (invariant: contenido embebido)</h2>
`
for (const [name, label] of MOCKUPS) {
  const doc = readFileSync(`${R}proposal/${name}.html`, 'utf8')
  html += `  <details open><summary>${label} — proposal/${name}.html</summary>
  <iframe srcdoc="${esc(doc)}" title="${name}"></iframe></details>\n`
}

html += `  <div class="transparency"><strong>Notas de transparencia:</strong>
  (1) P6 banner muted/30 y P7 banner offline son <em>demos inyectados</em> (marcados <code>data-demo-injected</code> en el mockup) — estados condicionales no reproducibles en vivo sin falla de red/borrador restaurado; markup = copia exacta del componente real con las clases propuestas. (2) Imágenes de producto en mocks dependen del dev server local activo. (3) El "después" de cada par es render del HTML de propuesta a viewport idéntico (fullPage, dsf=2), no screenshot de Stitch.
  </div>
</section>
</main>
</body>
</html>
`

writeFileSync(`${R}comparison.html`, html)
console.log('comparison.html written:', (html.length / 1e6).toFixed(1), 'MB')
